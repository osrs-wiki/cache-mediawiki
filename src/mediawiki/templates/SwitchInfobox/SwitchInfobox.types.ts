export type SwitchInfoboxItem = {
  text: string; // Label for the tab (e.g., "Royal servant", "In combat")
  item: string; // MediaWiki template content
};

export type SwitchInfoboxOptions = {
  items: SwitchInfoboxItem[];
};
