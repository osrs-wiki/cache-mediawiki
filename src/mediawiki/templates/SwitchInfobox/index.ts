export { SwitchInfobox } from "./SwitchInfobox";
export * from "./SwitchInfobox.types";
