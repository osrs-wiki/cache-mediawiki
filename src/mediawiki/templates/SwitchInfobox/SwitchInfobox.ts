import { MediaWikiTemplate } from "@osrs-wiki/mediawiki-builder";

import { SwitchInfoboxItem } from "./SwitchInfobox.types";
import { Template } from "../templates.types";

export class SwitchInfobox extends Template {
  items: SwitchInfoboxItem[];

  constructor(items: SwitchInfoboxItem[]) {
    super("Switch infobox");
    this.items = items;
  }

  build(): MediaWikiTemplate {
    const template = new MediaWikiTemplate(this.name);

    this.items.forEach((item, index) => {
      const num = index + 1;
      template.add(`text${num}`, item.text);
      template.add(`item${num}`, item.item);
    });

    return template;
  }
}
