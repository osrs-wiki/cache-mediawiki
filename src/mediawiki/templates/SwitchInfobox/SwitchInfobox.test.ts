import { SwitchInfobox } from "./SwitchInfobox";

describe("SwitchInfobox", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should build switch infobox with multiple items", () => {
    const items = [
      { text: "Royal servant", item: "{{Infobox NPC|name=Ennius Tullus}}" },
      { text: "In combat", item: "{{Infobox Monster|name=Ennius Tullus}}" },
    ];
    const template = new SwitchInfobox(items);
    const result = template.build();

    expect(result.name).toBe("Switch infobox");
    expect(result.build()).toMatchSnapshot();
  });

  it("should build switch infobox with three items", () => {
    const items = [
      { text: "Royal servant", item: "{{Infobox NPC|name=Ennius Tullus}}" },
      { text: "Cultist", item: "{{Infobox NPC|name=Ennius Tullus (cultist)}}" },
      { text: "In combat", item: "{{Infobox Monster|name=Ennius Tullus}}" },
    ];
    const template = new SwitchInfobox(items);
    const result = template.build();

    expect(result.name).toBe("Switch infobox");
    expect(result.build()).toMatchSnapshot();
  });

  it("should handle empty items array", () => {
    const items: { text: string; item: string }[] = [];
    const template = new SwitchInfobox(items);
    const result = template.build();

    expect(result.name).toBe("Switch infobox");
    expect(result.build()).toMatchSnapshot();
  });

  it("should handle single item", () => {
    const items = [{ text: "Default", item: "{{Infobox NPC|name=Test NPC}}" }];
    const template = new SwitchInfobox(items);
    const result = template.build();

    expect(result.name).toBe("Switch infobox");
    expect(result.build()).toMatchSnapshot();
  });
});
