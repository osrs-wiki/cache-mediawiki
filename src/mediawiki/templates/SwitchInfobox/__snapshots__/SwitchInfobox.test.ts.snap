// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchInfobox should build switch infobox with multiple items 1`] = `
"{{Switch infobox
|text1 = Royal servant
|item1 = {{Infobox NPC|name=Ennius Tullus}}
|text2 = In combat
|item2 = {{Infobox Monster|name=Ennius Tullus}}
}}
"
`;

exports[`SwitchInfobox should build switch infobox with three items 1`] = `
"{{Switch infobox
|text1 = Royal servant
|item1 = {{Infobox NPC|name=Ennius Tullus}}
|text2 = Cultist
|item2 = {{Infobox NPC|name=Ennius Tullus (cultist)}}
|text3 = In combat
|item3 = {{Infobox Monster|name=Ennius Tullus}}
}}
"
`;

exports[`SwitchInfobox should handle empty items array 1`] = `
"{{Switch infobox}}
"
`;

exports[`SwitchInfobox should handle single item 1`] = `
"{{Switch infobox|text1=Default|item1={{Infobox NPC|name=Test NPC}}}}
"
`;
